---
title: 'Analysis of SARS-CoV-2 SNP Data'
author: "Kim Medina"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: html_document
csl: bioinformatics.csl
bibliography: references.bib
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      warning = FALSE,
                      message = FALSE,
                      error = FALSE)
```

## Setup
```{r, assignment-of-SNP-dataset-and-sample-ID-to-variable}
snp_data <- read.csv("data/sample_snp_data.csv")
sample_id63 <- "SRR12433063"
sample_id65 <- "SRR12433065"
sample_id66 <- "SRR12433066"
```

```{r, load-packages-and-source-function-file}
library(dplyr)
library(ggplot2)
library(knitr)
source("code/functions/subset_data_function.R")
```

## Project Description

This project analyzes single nucleotide polymorphism (SNP) data from SARS-CoV-2 samples. It subsets the data to focus solely on one sample only. The SNP data is then visualized in a plot. The first package used is dplyr [@dplyr]. Then ggplot2 is used to create the plot [@ggplot2]. Lastly, knitr is used to create the report [@knitr].

## Subsetting data

```{r, writing-csv-file}
single_sample_data <- subset_snp_data(snp_data, sample_id63)
output <- (paste("output/sample_snp_data_", sample_id63, ".csv"))
write.csv(single_sample_data, file = output, row.names = FALSE)
kable(single_sample_data, caption = paste0("SNP data for sample ", sample_id63))
```

## Figure

```{r, plot-quality-across-genome-subset}
snp_data %>%
  subset_snp_data(sample_id63) %>%
  ggplot(aes(x = pos,
             y = qual,
             color = sample)) +
  geom_point(size = 3) +
  labs(title = paste0(
  "SNP quality of sample ", sample_id63, " across SARS-CoV-2 Genome"),
  x = "Position in Genome (nt)",
  y = "SNP Quality")

ggsave(paste0("output/plot_quality_across_genome_", sample_id63, ".png"))
```

### Session Info
```{r, session-info}
devtools::session_info()
```

# Sources Cited
